### PR TITLE
chore(github): update PR/issue templates

### PR DESCRIPTION
**Pull request by 0ubbe**
## Reasoning 💡

This is a proposition to improve our PR template by:
- adding emotion to it through emojis 🎨 
- rename sections 
- remove `"How"` section; I think the dev can use either `"Summary"` or `"Reasoning"` to explain this 
- add "Affected issues" section to hint issuers to find/link issues for better visibility

I'm using this PR description to illustrate how the "proposed" structure would feel like.

## Checklist 🧢

- [ ] ~Documentation~
- [ ] ~Tests~
- [ ] ~Ready to be merged~

## Affected issues 🎟

#1792

### CODE REVIEW

**Comment by balazsorban44** [comment]
Could we maybe have a look at the issue templates in this PR as well? Any good suggestions there?

I think as a minimum, we should demand reproductions for bug reports, and notice that without one, the issue is at risk of being closed. We HAVE to be stricter in order to force folks to read the docs, or if it turns out they are rightfully saying the docs is insufficient, we could act and update it where it can be instead of pushing it with a comment "this could be in the docs"

**Comment by balazsorban44** [comment]
I'd we wanted to make the "Affected issues" even more useful, we could hint that writing "Fixes" or "Closes" before the issue number will help with maintenance as it will automatically close such connected issues

**Comment by lluia** [comment]
@balazsorban44 ( cc. @StefanSelfTaught  @iaincollins ) I updated all the issue templates, preview here how they'll look:
- [Bug report 🐜 ](https://github.com/nextauthjs/next-auth/blob/39784afb1fd662b3ce24dbd29c9aa303c5771f10/.github/ISSUE_TEMPLATE/bug_report.md)
- [Feature request 🥑](https://github.com/nextauthjs/next-auth/blob/39784afb1fd662b3ce24dbd29c9aa303c5771f10/.github/ISSUE_TEMPLATE/feature_request.md)
- [Question ❓](https://github.com/nextauthjs/next-auth/blob/39784afb1fd662b3ce24dbd29c9aa303c5771f10/.github/ISSUE_TEMPLATE/question.md)

I removed some sections that I think were too verbose and tried to make them more friendly. I added some hints on how to document well each through code examples, screenshots, etc... I also added a **Contributing** section on each to hint people help themselves 🤞🏽

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -2,25 +2,33 @@
...
-**Additional context**
-Any other context, screenshots, etc.
+```js
+function myAwesomeNextAuthFeature() {
+  return 💚
+}
+```
 
-*Please indicate if you are willing and able to help implement the proposed feature.*
+Take time thinking what you want to say and helps us understand your proposal making sure that this description contains:
```
Comment (balazsorban44) [inline .github/ISSUE_TEMPLATE/feature_request.md:24]: help us*

**Review by balazsorban44 — APPROVED** [review]
This is awesome! Love it! could you just scan them for typos again?

Should we introduce a TypesScript issue template? Maybe similar to a question, only we won't convert them to discussions, and they will be auto labeled with TS?

It can be a separate PR though, so for me this is good to go

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1 @@
+/types/ @balazsorban44 @lluia
```
[the remark this answers is not in the document]
Comment (lluia) [reply .github/CODEOWNERS]: Yup, my bad I branched from my previous PR instead from `main` 🤣 , I'll `rebase` and remove the extraneous commits 👍🏽

**Comment by lluia** [comment]
@balazsorban44 created a **TS** template, fixed the typo you found and checked typos again 👍🏽

**Review by balazsorban44 — APPROVED** [review]

**Comment by balazsorban44** [comment]
LGTM!

**Pull request merged**